packages/xod-project/src/utils.js

```diff
@@ -146,3 +146,38 @@ export const resolveLinkNodeIds = R.curry((nodeIdMap, links) =>
     links
   )
 );
+
+// =============================================================================
+// RegExp parts for `isValidNumberDataValue`
+// =============================================================================
+
+// Optional sign
+const optSignRegExp = '(-|\\+)?';
+// Digits or digits+dot or dot+digits or digits+dot+digits
+const digitsWithDotRegExp = '(\\d+|\\d+\\.|\\.\\d+|\\d+\\.\\d+)';
+// Optional scientific notation with exponential part
+const scientificPartRegExp = '((e-|e\\+|e|\\d+)\\d+)?';
```

Comment:
Can’t get with an idea how it could hurt, but what `\\d+` in the inner parens does? Scientific notation always starts with `e`, isn’t it? And you have `optSignRegExp`, so shouldn’t it be `(e${optSignRegExp}\\d+)?}`?